File: src/main/java/com/thanabodee2661/springcommon/model/ResponseStatusModel.java
package com.thanabodee2661.springcommon.model;

import com.thanabodee2661.springcommon.enumeration.ResponseStatusEnum;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
public class ResponseStatusModel {
    private String code;
    private String message;
    private String description;

    public ResponseStatusModel(ResponseStatusEnum statusEnum) {
        this.code = statusEnum.getCode();
        this.message = statusEnum.getMessage();
        this.description = statusEnum.getDescription();
    }
}

File: src/main/kotlin/jabza/com/common/model/ResponseStatusModel.kt
package jabza.com.common.model;

import jabza.com.common.enumeration.ResponseStatusEnum

data class ResponseStatusModel (val code: String, val message: String, val description: String) {
    constructor(statusEnum: ResponseStatusEnum): this(statusEnum.code, statusEnum.message, statusEnum.description)
}
